pkg/credentials/credentials.go

```diff
@@ -283,13 +287,45 @@ func defaultValidationFunc(creds map[string]string) error {
 }
 
 func openstackValidationFunc(creds map[string]string) error {
-	for k, v := range creds {
-		if k == OpenStackTenantID || k == OpenStackTenantName {
-			continue
+	alwaysRequired := []string{OpenStackAuthURL, OpenStackDomainName, OpenStackRegionName}
```

Comment:
@kron4eg Do you have some idea if the Domain Name is really required? I'm trying to figure it out from the [GopherCloud library](https://github.com/gophercloud/gophercloud/blob/master/openstack/auth_env.go), and it seems that they only require it if `applicationCredentialID` is empty and `applicationCredentialSecret` is not empty, which can't happen in our case.

Instead, they have a validation that Tenant ID is required if the Tenant Name is provided, and domain ID and name are not provided.

Reply:
The list of "always required" is basically just subtracting the variables that are checked separately from the complete list of variables, so this was required before, but I'm open to removing it.

Reply:
I guess at this moment we don't have a choice

https://github.com/kubermatic/machine-controller/blob/07e3ca3aa199ca8ada73fada539ef565b45612a1/pkg/cloudprovider/provider/openstack/provider.go#L444-L446

Reply:
Good point, I forgot to check in machine-controller.